src/test/ui/span/borrowck-ref-into-rvalue.nll.stderr

```diff
@@ -1,16 +1,16 @@
-error[E0597]: borrowed value does not live long enough
-  --> $DIR/borrowck-ref-into-rvalue.rs:13:11
+error[E0713]: borrow may still be in use when destructor runs
+  --> $DIR/borrowck-ref-into-rvalue.rs:14:14
    |
-LL |     match Some("Hello".to_string()) {
-   |           ^^^^^^^^^^^^^^^^^^^^^^^^^ temporary value does not live long enough
+LL |         Some(ref m) => {
+   |              ^^^^^
 ...
 LL |     }
-   |     - temporary value only lives until here
+   |     - drop of temporary value occurs here
```

Comment:
hmm I have to admit that this delta (namely losing the printout of the `match`'s input expression) seems unfortunate. I should look into incorporating that into the diagnostic output in cases like this.

Reply:
Musing on this, I was thinking that it might be a good idea for us to explain *why* the value is being dropped here. One way I could imagine doing that is:

- First, find the thing that was borrowed. If it is a user-local, then add a note with that span and say something like:

```
= note: local variables are dropped when they go out of scope

    let x;
          - `x` declared here
  }
  - `x` goes out of scope here
```

note sure if we can do notes with multispans like that (cc @estebank) but it seems ideal. Alternatively, we could merge this into the main error.

Reply:
For temporaries, we would probably give a slightly different error, along the lines of what @mikhail-m1  did in https://github.com/rust-lang/rust/pull/54164/

Reply:
@nikomatsakis not supported yet, but something I've had in my backlog for a while. Should I prioritize it?

Reply:
@estebank I personally don't think you need to re-prioritize. We can work around its absence, as @nikomatsakis notes...